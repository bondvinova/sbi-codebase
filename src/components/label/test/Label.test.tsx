import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Label from '@/components/label';

describe('Label component', () => {
  it('Should render and match the snapshot', () => {
    const { baseElement } = render(<Label>Label</Label>);

    expect(baseElement).toMatchSnapshot();
  });

  it('Should render correct ui with different props', () => {
    render(
      <Label variant="select" size="base" htmlFor="Username">
        Label
      </Label>
    );

    const result = screen.getByText('Label');

    expect(result).toBeInTheDocument();
    expect(result).toHaveClass('text-grey-slate-110');
    expect(result).toHaveClass('text-base leading-6');
  });
});
