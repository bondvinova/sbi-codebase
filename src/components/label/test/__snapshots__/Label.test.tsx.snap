// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Label component > Should render and match the snapshot 1`] = `
<body>
  <div>
    <label
      class="peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-normal text-global-slate-80 text-sm leading-5"
    >
      Label
    </label>
  </div>
</body>
`;
